### PR TITLE
docs: add gst capi face detection documentation

### PR DESCRIPTION
**Pull request by vli11**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-self-checkout/blob/main/CONTRIBUTING.md
- [ ] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [x] Updated Documentation as relevant to the changes
- [x] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?
add gst capi face detection documentation

## Issue this PR will close

close: #**issue_number**

### CODE REVIEW

File: docs_src/OVMS/capiPipelineRun.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Openvino OVMS C-API Pipeline Run
+
+Openvino has [many ways to run inferencing pipeline](https://docs.openvino.ai/2023.1/ovms_docs_server_api.html):
```
Comment (jim-wang-intel) [inline docs_src/OVMS/capiPipelineRun.md:3]:
```suggestion
OpenVino has [many ways to run inferencing pipeline](https://docs.openvino.ai/2023.1/ovms_docs_server_api.html): 
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+# Openvino OVMS C-API Pipeline Run
+
+Openvino has [many ways to run inferencing pipeline](https://docs.openvino.ai/2023.1/ovms_docs_server_api.html): 
+TensorFlow Serving gRPC API, KServe gRPC API, TensorFlow Serving REST API, KServe REST API and OVMS C API through model server. Here is a demonstration for using OVMS C API method to run face detection inferencing pipeline with steps below:
```
Comment (jim-wang-intel) [inline docs_src/OVMS/capiPipelineRun.md:4]:
```suggestion
TensorFlow Serving gRPC API, KServe gRPC API, TensorFlow Serving REST API, KServe REST API and C API through OpenVino model server (OVMS). Here is a demonstration for using OVMS C API method to run face detection inferencing pipeline with steps below:
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+1. Add new section to model configuration file for model server
+2. Add environment variable file dependency
+3. add a profile launcher pipeline configuration file
+4. build and run
+
+
+## Add New Section To Model Config File for Model Server
+Here is the config file location: `configs/opencv-ovms/models/2022/config_template.json`, edit the file and append the following face detection model configuration section to the template
```
Comment (jim-wang-intel) [inline docs_src/OVMS/capiPipelineRun.md:13]:
```suggestion
Here is the template config file location: `configs/opencv-ovms/models/2022/config_template.json`, edit the file and append the following face detection model configuration section to the template
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      "plugin_config": {"PERFORMANCE_HINT": "LATENCY"},
+      "target_device": "{target_device}"},
+      "latest": { "num_versions": 2 }
+    }
+```
+!!! Note
+    `shape` is optional and takes precedence over batch_size, please remove this attribute if you don't know the value for the model.
+
+!!! Note
+    Please leave `target_device` value as it is, as the value `{target_device}` will be recognized and filled out by script run.
```
Comment (jim-wang-intel) [inline docs_src/OVMS/capiPipelineRun.md:31]:
```suggestion
    Please leave `target_device` value as it is, as the value `{target_device}` will be recognized and replaced by script run.
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+You can find each parameter description from model server [ovms_docs_parameters](https://docs.openvino.ai/2023.1/ovms_docs_parameters.html).
```
Comment (jim-wang-intel) [inline docs_src/OVMS/capiPipelineRun.md:33]:
```suggestion
You can find the parameter description from model server [ovms_docs_parameters](https://docs.openvino.ai/2023.1/ovms_docs_parameters.html).
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+## Add Environment Variable File
+You can add environment variable file to `configs/opencv-ovms/envs/` directory and you can add more than one environment variable file for your pipeline. For face detection pipeline run, we have added `configs/opencv-ovms/envs/capi_face_detection.env`, below is a list of explanation for each environment variable and current default values we set for face detection pipeline run, this list can be extended for any future modification.
```
Comment (jim-wang-intel) [inline docs_src/OVMS/capiPipelineRun.md:36]:
```suggestion
You can add multiple environment variable files to `configs/opencv-ovms/envs/` directory for your pipeline. For face detection pipeline run, we have added `configs/opencv-ovms/envs/capi_face_detection.env` environment variable file,. Below is a list of explanation for all environment variables and current default values we set for face detection pipeline run, this list can be extended for any future modification.
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+| EV Name                   | Description                                           |
+| --------------------------| ------------------------------------------------------|
+| RENDER_PORTRAIT_MODE      | rendering in portrait mode                            |
+| GST_DEBUG                 | GStreamer debug mode option                           |
+| USE_ONEVPL                | using OneVPL CPU & GPU Support                        |
+| PIPELINE_EXEC_PATH        | pipeline execution path inside container              |
+| GST_VAAPI_DRM_DEVICE      | GStreamer VAAP DRM device input                       |
```
Comment (jim-wang-intel) [inline docs_src/OVMS/capiPipelineRun.md:44]:
```suggestion
| GST_VAAPI_DRM_DEVICE      | GStreamer VAAPI DRM device input                       |
```

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
suggestion for some wording changes

File: docs_src/OVMS/capiPipelineRun.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  PipelineInputArgs: "" # space delimited like we run the script in command and take those input arguments
+  EnvironmentVariableFiles:
+    - capi_face_detection.env
+```
+
+## Build and Run
+1. Build gst-capi ovms with profile-launcher: `make build-gst-capi`
+2. Download sample video files: `cd benchmark-scripts/ && ./download_sample_videos.sh && cd ..`
+2. Start simulator camera: `./camera-simulator/camera-simulator.sh`
+3. To start face detection pipeline: `PIPELINE_PROFILE="capi_face_detection" RENDER_MODE=1 sudo -E ./run.sh --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0 --workload ovms`
```
Comment (jim-wang-intel) [inline docs_src/OVMS/capiPipelineRun.md:89]:
```suggestion
4. To start face detection pipeline: `PIPELINE_PROFILE="capi_face_detection" RENDER_MODE=1 sudo -E ./run.sh --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0 --workload ovms`
```

File: docs_src/OVMS/capiPipelineRun.md (hunk 1 above)
Comment (NeethuES-intel) [reply docs_src/OVMS/capiPipelineRun.md:3]: Actual annotation is OpenVINO, do we want to change it everywhere to be consistent ?

File: docs_src/OVMS/capiPipelineRun.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
+# Openvino OVMS C-API Pipeline Run
+
+Openvino has [many ways to run inferencing pipeline](https://docs.openvino.ai/2023.1/ovms_docs_server_api.html): 
+TensorFlow Serving gRPC API, KServe gRPC API, TensorFlow Serving REST API, KServe REST API and OVMS C API through model server. Here is a demonstration for using OVMS C API method to run face detection inferencing pipeline with steps below:
+
+1. Add new section to model configuration file for model server
+2. Add environment variable file dependency
+3. add a profile launcher pipeline configuration file
```
Comment (NeethuES-intel) [inline docs_src/OVMS/capiPipelineRun.md:8]:
```suggestion
3. Add a profile launcher pipeline configuration file
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+4. build and run
```
Comment (NeethuES-intel) [inline docs_src/OVMS/capiPipelineRun.md:9]:
```suggestion
4. Build and run
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+| EV Name                   | Description                                           |
+| --------------------------| ------------------------------------------------------|
+| RENDER_PORTRAIT_MODE      | rendering in portrait mode                            |
+| GST_DEBUG                 | GStreamer debug mode option                           |
+| USE_ONEVPL                | using OneVPL CPU & GPU Support                        |
+| PIPELINE_EXEC_PATH        | pipeline execution path inside container              |
+| GST_VAAPI_DRM_DEVICE      | GStreamer VAAP DRM device input                       |
+| TARGET_GPU_DEVICE         | target GPU device                                     |
+| LOG_LEVEL                 | pipeline run log level option                         |
```
[the remark this answers is not in the document]
Comment (ejlee3) [reply docs_src/OVMS/capiPipelineRun.md:46]: maybe list the possible values - DEBUG, INFO, WARNING, ERROR (or whatever the levels are)

File: docs_src/OVMS/capiPipelineRun.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+| EV Name                   | Description                                           |
+| --------------------------| ------------------------------------------------------|
+| RENDER_PORTRAIT_MODE      | value 1 for rendering in portrait mode                |
+| GST_DEBUG                 | value 1 for running GStreamer in debug mode           |
+| USE_ONEVPL                | value 1 for using OneVPL CPU & GPU Support            |
+| PIPELINE_EXEC_PATH        | pipeline execution path inside container              |
+| GST_VAAPI_DRM_DEVICE      | GStreamer VAAPI DRM device input                      |
+| TARGET_GPU_DEVICE         | privilege mode for using GPU devices if any           |
+| LOG_LEVEL                 | [log level](https://gstreamer.freedesktop.org/documentation/tutorials/basic/debugging-tools.html?gi-language=c#the-debug-log) to be set when running gst pipeline         |
```
Comment (jim-wang-intel) [inline docs_src/OVMS/capiPipelineRun.md:46]:
```suggestion
| LOG_LEVEL                 | [GST_DEBUG log level](https://gstreamer.freedesktop.org/documentation/tutorials/basic/debugging-tools.html?gi-language=c#the-debug-log) to be set when running gst pipeline         |
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+  PipelineInputArgs: "" # space delimited like we run the script in command and take those input arguments
+  EnvironmentVariableFiles:
+    - capi_face_detection.env
+```
+
+## Build and Run
+Here are the quick start steps to build and run OVMS C API face detection pipeline profile:
+1. Build gst-capi ovms with profile-launcher: `make build-gst-capi`
+2. Download sample video files: `cd benchmark-scripts/ && ./download_sample_videos.sh && cd ..`
+3. Start simulator camera: `make run-camera-simulator`
```
Comment (jim-wang-intel) [inline docs_src/OVMS/capiPipelineRun.md:89]:
```suggestion
## Build and Run

Here are the quick start steps to build and run OVMS C API face detection pipeline profile:

1. Build gst-capi ovms with profile-launcher: `make build-gst-capi`
2. Download sample video files: `cd benchmark-scripts/ && ./download_sample_videos.sh && cd ..`
3. Start simulator camera: `make run-camera-simulator`
```
please insert a new line for between title and the first sentence and then also a new line before bullets or number in order for mkdocs generation properly

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

File: docs_src/OVMS/capiPipelineRun.md [hunk 14]
```diff
@@ -28,23 +28,23 @@ Here is the config file location: `configs/opencv-ovms/models/2022/config_templa
...
 
-You can find each parameter description from model server [ovms_docs_parameters](https://docs.openvino.ai/2023.1/ovms_docs_parameters.html).
+You can find the parameter description from model server [ovms_docs_parameters](https://docs.openvino.ai/2023.1/ovms_docs_parameters.html).
 
 ## Add Environment Variable File
-You can add environment variable file to `configs/opencv-ovms/envs/` directory and you can add more than one environment variable file for your pipeline. For face detection pipeline run, we have added `configs/opencv-ovms/envs/capi_face_detection.env`, below is a list of explanation for each environment variable and current default values we set for face detection pipeline run, this list can be extended for any future modification.
+You can add mulitple environment variable files to `configs/opencv-ovms/envs/` directory for your pipeline. For face detection pipeline run, we have added `configs/opencv-ovms/envs/capi_face_detection.env` environment variable file. Below is a list of explanation for all environment variables and current default values we set for face detection pipeline run, this list can be extended for any future modification.
```
Comment (ejlee3) [inline docs_src/OVMS/capiPipelineRun.md:36]:
```suggestion
You can add multiple environment variable files to `configs/opencv-ovms/envs/` directory for your pipeline. For face detection pipeline run, we have added `configs/opencv-ovms/envs/capi_face_detection.env` environment variable file. Below is a list of explanation for all environment variables and current default values we set for face detection pipeline run, this list can be extended for any future modification.
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 15]
```diff
@@ -28,23 +28,23 @@ Here is the config file location: `configs/opencv-ovms/models/2022/config_templa
     `shape` is optional and takes precedence over batch_size, please remove this attribute if you don't know the value for the model.
 
 !!! Note
-    Please leave `target_device` value as it is, as the value `{target_device}` will be recognized and filled out by script run.
+    Please leave `target_device` value as it is, as the value `{target_device}` will be recognized and replaced by script run.
 
-You can find each parameter description from model server [ovms_docs_parameters](https://docs.openvino.ai/2023.1/ovms_docs_parameters.html).
+You can find the parameter description from model server [ovms_docs_parameters](https://docs.openvino.ai/2023.1/ovms_docs_parameters.html).
```
Comment (ejlee3) [inline docs_src/OVMS/capiPipelineRun.md:33]:
```suggestion
You can find the parameter description from in the [ovms docs](https://docs.openvino.ai/2023.1/ovms_docs_parameters.html).
```

File: docs_src/OVMS/capiPipelineRun.md [hunk 16]
```diff
@@ -28,23 +28,23 @@ Here is the config file location: `configs/opencv-ovms/models/2022/config_templa
...
+You can add mulitple environment variable files to `configs/opencv-ovms/envs/` directory for your pipeline. For face detection pipeline run, we have added `configs/opencv-ovms/envs/capi_face_detection.env` environment variable file. Below is a list of explanation for all environment variables and current default values we set for face detection pipeline run, this list can be extended for any future modification.
 
 | EV Name                   | Description                                           |
 | --------------------------| ------------------------------------------------------|
-| RENDER_PORTRAIT_MODE      | rendering in portrait mode                            |
-| GST_DEBUG                 | GStreamer debug mode option                           |
-| USE_ONEVPL                | using OneVPL CPU & GPU Support                        |
+| RENDER_PORTRAIT_MODE      | value 1 for rendering in portrait mode                |
```
Comment (ejlee3) [inline docs_src/OVMS/capiPipelineRun.md:40]: Can we split this table into 3 columns to make it easier to read? 

| EV Name   | Default Value   | Description   |

File: docs_src/OVMS/capiPipelineRun.md [hunk 17]
```diff
@@ -0,0 +1,92 @@
+# OpenVINO OVMS C-API Pipeline Run
+
+OpenVINO has [many ways to run inferencing pipeline](https://docs.openvino.ai/2023.1/ovms_docs_server_api.html):
```
Comment (antoniomtz) [inline docs_src/OVMS/capiPipelineRun.md:3]:
```suggestion
OpenVINO Model Server has [many ways to run inferencing pipeline](https://docs.openvino.ai/2023.1/ovms_docs_server_api.html): 
```

**Review by antoniomtz — CHANGES_REQUESTED** [review]

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by ejlee3 — APPROVED** [review]
LGTM

**Review by antoniomtz — APPROVED** [review]
LGTM

**Pull request merged**